codec_parameters.go

```diff
@@ -141,6 +145,27 @@ func (cp *CodecParameters) SetSampleAspectRatio(r Rational) {
 	cp.c.sample_aspect_ratio = r.c
 }
 
+func (cp *CodecParameters) SetExtraData(extraData []byte) error {
+	if len(extraData) == 0 {
+		return nil
+	}
+
+	if cp.c.extradata != nil {
+		C.av_freep(unsafe.Pointer(&cp.c.extradata))
+		cp.c.extradata_size = 0
+	}
+
+	extradataSize := len(extraData)
+	cp.c.extradata = (*C.uint8_t)(C.av_mallocz(C.size_t(extradataSize + C.AV_INPUT_BUFFER_PADDING_SIZE)))
+	if cp.c.extradata == nil {
+		return fmt.Errorf("failed to allocate extradata")
+	}
+
+	C.memcpy(unsafe.Pointer(cp.c.extradata), unsafe.Pointer(&extraData[0]), C.size_t(extradataSize))
+	cp.c.extradata_size = C.int(extradataSize)
+	return nil
+}
+
```

Comment:
Could you add a `func (cp *CodecParameters) ExtraData() ([]byte, error)` method as well (using `bytesFromC` like [here](https://github.com/asticode/go-astiav/blob/master/packet.go#L89))?